boards/posix/native_sim/doc/index.rst

```diff
@@ -567,13 +567,16 @@ Interaction with serial ports can be configured in several different ways:
   options override values from the devicetree.
 * The rest of the configuration options such as number of data and stop bits,
   parity, as well as baud rate can be set at runtime with ``uart_configure``.
+* This driver can emulate an interrupt-driven UART by enabling
+  :kconfig:option:`CONFIG_UART_INTERRUPT_DRIVEN`.
 
 Multiple instances of such uart drivers are supported.
 
 The :zephyr:code-sample:`uart-native-tty` sample app provides a working example of the
 driver.
 
-This driver only supports poll mode. Interrupt and async mode are not supported.
+This driver only supports poll mode and interrupt mode. Async mode is not
+supported.
```

Comment:
I think the first paragraph of this section now also requires updating. It currently says "With this driver an application can use the polling UART API (uart_poll_out, uart_poll_in) to write and read characters to and from a connected serial port device."
cc @aescolar

Reply:
BTW, would it make sense / be possible to have the same approach for the uart_native_pty driver?

Reply:
@kartben Sure, I guess nobody though it worthwhile so far